src/mds/Locker.cc

```diff
@@ -2991,6 +2992,10 @@ void Locker::_update_cap_fields(CInode *in, int dirty, MClientCaps *m, inode_t *
 	      << " for " << *in << dendl;
       pi->mtime = mtime;
     }
+    if (((dirty & CEPH_CAP_FILE_WR) && change_attr > pi->change_attr) ||
+	((dirty & CEPH_CAP_FILE_EXCL) && change_attr != pi->change_attr)) {
+      pi->change_attr = change_attr;
```

Comment:
Hmm...what I think I need to do is sit down and write this up in better detail. I think this is OK, but the reason for that is somewhat subtle. Also, given that this PR is closed and that the new one is now open, it's probably better to continue the discussion there.